lesson_5/lesson_5.py

```diff
@@ -0,0 +1,139 @@
+  'Tesla': ('silver', 2020, 0.0, 'sedan', 60000),
+  'Acura': ('white', 2017, 2.4, 'suv', 40000),
+  'Cadillac': ('black', 2019, 3.6, 'suv', 55000),
+  'Infiniti': ('gray', 2018, 2.0, 'sedan', 35000),
+  'Lincoln': ('white', 2021, 2.0, 'suv', 50000),
+  'GMC': ('blue', 2016, 1.5, 'pickup', 30000),
+  'Ram': ('black', 2019, 5.7, 'pickup', 40000),
+  'Chevy': ('red', 2017, 2.4, 'pickup', 35000),
+  'Dodge Ram': ('white', 2020, 3.6, 'pickup', 45000),
+  'Ford F-Series': ('gray', 2021, 3.5, 'pickup', 50000),
+  'Nissan Titan': ('silver', 2018, 5.6, 'pickup', 35000)
+}
+
+search_criteria = (2017, 1.6, 36000)
+
+
+def filter_cars(car_data, search_criteria):
+    filtered_cars = []
+    year_min, engine_volume_min, price_max = search_criteria
+
+    for car, attributes in car_data.items():
+        color, year, engine_volume, car_type, price = attributes
+        if year >= year_min and engine_volume >= engine_volume_min and price <= price_max:
+            filtered_cars.append((car, attributes))
+
+    filtered_cars.sort(key=lambda x: x[1][4])
+
+    return filtered_cars[:5]
+
+
+filtered_cars = filter_cars(car_data, search_criteria)
+
+print("Task 5.1")
+
+for car in filtered_cars:
+    print(car)
+
+
+# Given list of tuples (name, surname, age, profession, City location)
+# 1 - Add your new record o the beginning of the given list
+# 2 - In modified list swap elements with indexes 1 and 5 (1<->5). Print result
+# 3 - check that all people in modified list with records indexes 6, 10, 13
+#   have age >=30. Print condition check result
+
+people_records = [
+  ('John', 'Doe', 28, 'Engineer', 'New York'),
+  ('Alice', 'Smith', 35, 'Teacher', 'Los Angeles'),
+  ('Bob', 'Johnson', 45, 'Doctor', 'Chicago'),
+  ('Emily', 'Williams', 30, 'Artist', 'San Francisco'),
+  ('Michael', 'Brown', 22, 'Student', 'Seattle'),
+  ('Sophia', 'Davis', 40, 'Lawyer', 'Boston'),
+  ('David', 'Miller', 33, 'Software Developer', 'Austin'),
+  ('Olivia', 'Wilson', 27, 'Marketing Specialist', 'Denver'),
+  ('Daniel', 'Taylor', 38, 'Architect', 'Portland'),
+  ('Grace', 'Moore', 25, 'Graphic Designer', 'Miami'),
+  ('Samuel', 'Jones', 50, 'Business Consultant', 'Atlanta'),
+  ('Emma', 'Hall', 31, 'Chef', 'Dallas'),
+  ('William', 'Clark', 29, 'Financial Analyst', 'Houston'),
+  ('Ava', 'White', 42, 'Journalist', 'San Diego'),
+  ('Ethan', 'Anderson', 36, 'Product Manager', 'Phoenix')
+]
+
+print ('\n'
+       'Task 5.2')
+#my_list = [('Vitalii', 'Neskorodiev', 38, 'QA Engineer', 'Khmelnytskyy')]
+people_records.insert(0, ('Vitalii', 'Neskorodiev', 38, 'QA Engineer', 'Khmelnytskyy'))
+print(people_records)
+
+popped_element_1 = people_records.pop(1)
+popped_element_5 = people_records.pop(4)
+
+people_records.insert(0, popped_element_5)
+people_records.insert(4, popped_element_1)
+
+print(people_records)
+
+people_records_age_check = [people_records[6], people_records[10], people_records[13]]
+print(people_records_age_check)
+
+def age_check(person_age):
```

Comment:
You are doing double work with complicated solution. You can use only list comprehension if one line for check that persons are more or equal to 30.